std/algorithm/searching.d

```diff
@@ -1768,14 +1768,14 @@ if (isForwardRange!R1 && isForwardRange!R2
 
 /// ditto
 R1 find(alias pred = "a == b", R1, R2)(R1 haystack, R2 needle)
-if (isRandomAccessRange!R1 && isBidirectionalRange!R2
+if (isRandomAccessRange!R1 && hasLength!R1 && hasSlicing!R1 && isBidirectionalRange!R2
         && is(typeof(binaryFun!pred(haystack.front, needle.front)) : bool))
```

Comment:
It's not considered a breaking change, as this code is broken :)

Without these constraints, the compiler will allow types through that don't specify `length` or `opSlice`. Many people forget that's it's totally legal to have an infinite random access range, which this code currently, wrongly, accepts. Without these constraints, incorrect calls will result in the wrong error message, i.e. the "Does not have X method" rather than, "Does not satisfy any overload".